include/posix/sys/types.h

```diff
@@ -76,6 +76,21 @@ typedef struct pthread_rwlock_obj {
 	k_tid_t wr_owner;
 } pthread_rwlock_t;
 
+typedef u32_t pthread_once_t;
+
+/* pthread_key */
+typedef void *pthread_key_t;
+
+typedef struct pthread_key_obj {
```

Comment:
In types.h file, we do have similar `pthread_rwlock_obj` implementation detail, and I am not sure where else I would be adding this, since at least two other functions (pthread_key APIs and pthread_exit) are making use of `pthread_thread_data` and `pthread_key_obj`; and so I cannot add it to the pthread_key.c file. So could you help me here? @lpereira @nashif

Reply:
You can create header files that are included only by the implementation files.  These don't need to be in user-exposed headers.

Reply:
Okay sure @lpereira I will do that, thanks